### ISSUE TITLE
Allow use for 'var _' to be used to discard return values without raisin warning 'RETURN_VALUE_DISCARDED'

### ISSUE THREAD

**Issue by cxlpt** [godotengine/godot-proposals#5870]
### Describe the project you are working on

A top down 2D rpg game about a prison

### Describe the problem or limitation you are having in your project

In almost all of my projects, I invoke a function that has a return value I don't want to use (typically a godot api function such as 'move_and_slide', and so I just invoke the function without assigning the return value to a variable ('move_and_slide(input_vector * delta') rather than 'var x = move_and_slide(input_vector * delta)'). In doing so, I raise the warning 'RETURN_VALUE_DISCARDED'. However, if I assign the value to a variable, I don't use it, thus raising the error 'UNUSED_VARIABLE', which doesn't help.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allowing the use of an underscore to 'void' return results will create a concise syntax to say 'it is intentional I'm not using this return value, so don't raise the error'.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

```
move_and_slide(input_vector * delta) # raises the warning 'UNUSED_RETURN_RESULT'
var x =  move_and_slide(input_vector * delta) # raises the warning 'UNUSED_VARIABLE'
var _ =  move_and_slide(input_vector * delta) # would raise no warning
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

The three current solutions are the following: prefix the variable name with an underscore - this solution is fine, however, if I'm prefixing it with an underscore, that means I'm probably not going to use it, so why bother giving it a name rather than '_'; the second option is to mute the error globally, however, I believe this should never be done, as then what's the point in raising the warning; and third silence the error in that one instance using '#warning-ignore:warning_id', however putting this after every 'connect()' call is a rather untidy solution and feels unnatural.

### Is there a reason why this should be core and not an add-on in the asset library?

This (to my knowledge) cannot be achieved with an add-on. It is also an unobtrusive feature, as those who don't want to use it can just not use it, and is a problem that many people seem to encounter.

**Comment by YuriSizov** [comment]
See also https://github.com/godotengine/godot-proposals/issues/682. We've disabled the `RETURN_VALUE_DISCARDED` warning by default, by the way https://github.com/godotengine/godot/pull/69002 because of this exact issue.

**Comment by Mickeon** [comment]
> second option is to mute the error globally, however, I believe this should never be done, as then what's the point in raising the warning;

Indeed, it has been disabled by default for now (with a decent amount of positivity behind the decision, may I add), until a more proper warning is implemented.

In one RocketChat discussion, one idea discussed was to only raise a warning _(either this one or a new one)_ for `const` functions.  These functions do not modify anything, so discarding their return value is pretty much guaranteed to be an error on the programmer's part.

**Comment by geekley** [comment]
+1, but I'd like to propose better syntax for this. Preferably add both of these below.
Suppose `f(...)` is the method in question, where we don't necessarily need to use its return value.

Firstly, the proposed ***discard statement*** syntax shouldn't need `var`. It's unnecessarily verbose, as `_` can't even be used as an identifier name anyway. Like how it works in C#, it should be enough to do:
```gdscript
func some_function() -> void:
  _ = f(x) # will not raise the warning
```
You can think of this `_` semantics as if there was an implied "dummy" variable, that's always defined in any scope.

Second, I'd like to also have a ***discard expression*** syntax:
```gdscript
func some_function() -> void:
  void(f(x), f(y), f(z)) # no-op, all parameters are ignored
  #void f(x) # if going with Javascript syntax
```
This is the Javascript way, where the keyword `void` can be used like a no-op method that just ignores all the parameters and returns void (undefined in JS). You can think of it as if there was a global function like:
```gdscript
# if we go with this implementation, the return value is null and can be used
func void(_ignore: ...Variant) -> Nil:
  return null # no-op
# implementation where trying to use the return value raises error at compile time (arguably less useful?)
func void(_ignore: ...Variant) -> void:
  return # no-op
```
Technically, what JS allows is `void expression`, which is succinct, but personally I think it's quite confusing.
So I'd rather have just `void(expressions...)` instead, plus `_ = expression` for succinctness.

**Comment by outfrost** [comment]
@geekley What's the intended use case of `void(expr...)`? From the pov of this proposal, it seems moot to have a 2nd syntactic way to do the same thing, especially one that I think will be less intuitive to understand for a significant number of people without in-depth JS experience. To me as an experienced C/C++ dev, doing `void(f(x))` looks like either constructing a weird `void` type object, or calling a `void()` function, for an unclear, possibly cursed purpose. `_ = f(x)` I think communicates the intent much better.

**Comment by geekley** [comment]
Searching the internet for practical use cases of `void` operator/expression in JS, indeed it seems the only uses I find are specific to web and JavaScript's quirks. So that's a very valid point @outfrost. Specially considering GDScript syntax is not very expression-oriented (e.g. it lacks `++` operator, forcing you to do `+= 1`, which is a statement, not an expression).

So I agree that `_ = f(x)` is enough, just please drop the `var`.

**Comment by AleryBerry** [comment]
Is there any updates to this?

**Comment by geekley** [comment]
I'd like to suggest the syntax also be used to suppress `MISSING_AWAIT` (see linked issue above).
```gdscript
task_returns() # raises MISSING_AWAIT
var value := await task_returns() # OK
void_task() # raises MISSING_AWAIT
await void_task() # OK
_ = void_task() # Also OK, we don't want to await, just launch
_ = task_returns() # Also OK, we don't want to await, just launch
```
I.e. the "returned awaitable signal" of a function could also be ignored on purpose with the same syntax, even if it's an async `void` function. In fact, if you use it even on a sync `void` function, it shouldn't be an error. You're ignoring whatever is returned, nothing is returned anyway, no problem.

**Comment by MathyV** [comment]
In addition, I'd like to suggest that using `_` also silences the UNTYPED_DECLARATION warning. If you don't intend to do anything with the return value, it also doesn't make a lot of sense to do type checking on it.

```
var _discard: bool = move_and_slide()
```
vs
```
_ = move_and_slide()
```
is a clear win to me.
